apps/server/src/modules/tool/external-tool/controller/dto/request/external-tool-medium.params.ts

```diff
@@ -1,21 +1,36 @@
 import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
-import { IsNotEmpty, IsOptional, IsString } from 'class-validator';
+import { IsDate, IsEnum, IsNotEmpty, IsOptional, IsString } from 'class-validator';
+import { MediaSourceDataFormat } from '@modules/media-source';
 
 export class ExternalToolMediumParams {
 	@IsString()
 	@IsNotEmpty()
 	@ApiProperty({ type: String, description: 'Id of the medium' })
-	mediumId!: string;
+	public mediumId!: string;
 
 	@IsString()
 	@IsOptional()
 	@IsNotEmpty()
 	@ApiPropertyOptional({ type: String, description: 'Publisher of the medium' })
-	publisher?: string;
+	public publisher?: string;
 
 	@IsString()
 	@IsOptional()
 	@IsNotEmpty()
 	@ApiPropertyOptional({ type: String, description: 'The id of the media source' })
-	mediaSourceId?: string;
+	public mediaSourceId?: string;
+
+	@IsOptional()
+	@IsEnum(MediaSourceDataFormat, { each: true })
+	@ApiPropertyOptional({
+		enum: MediaSourceDataFormat,
+		enumName: 'MediaSourceDataFormat',
+		description: 'Restrict media sources to a specific format',
+	})
+	public format?: MediaSourceDataFormat;
+
+	@IsDate()
```

Comment:
the external tool does also not save the modified date of a media source

Reply:
Sorry I mean the modifiedAt date is not yet saved to the external tool so it should not be included. To be honest we should not include modifiedAt yet, since its purpose and use are not fully planned.